### PR TITLE
Fix VLANs and LAGs

### PR DESCRIPTION
**Pull request by milan-zededa**
Support for VLANs and LAGs has been added to EVE quite some time ago, but it has not been retested since. As we are soon planning to productize this feature, this PR contains a set of patches for regressions found while testing EVE with VLAN and LAG configurations.

Patches are mostly about skipping L2-only ports and not trying to use them for connectivity tests and HTTP requests.

I have also prepared an example for eden: https://github.com/lf-edge/eden/pull/939
Hopefully in the near future we will be able to turn this into a fully automated test.

### CODE REVIEW

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 1]
```diff
@@ -1684,29 +1684,6 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 		status.UUIDandVersion, status.DisplayName)
 }
 
-// VLAN filtering could have been enabled on the bridge immediately after
```
Comment (eriknordmark) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:1687]: Is this no longer needed due to some unknown netlink code? (New kernel version??)

**Review by eriknordmark — APPROVED** [review]
LGTM

File: pkg/pillar/cmd/domainmgr/domainmgr.go (hunk 1 above)
Comment (milan-zededa) [reply pkg/pillar/cmd/domainmgr/domainmgr.go:1687]: Some time ago, this was moved from domainmgr to zedrouter (NI Reconciler): https://github.com/lf-edge/eve/blob/master/pkg/pillar/nireconciler/linuxitems/vlanbridge.go#L106-L131
I just forgot to remove this no longer used function back then.

**Review by uncleDecart — APPROVED** [review]
LGTM

**Pull request merged**